src/recursion.erl

```diff
@@ -0,0 +1,33 @@
+-module(recursion).
+
+-export([recurse/1, fold/1, comprehension/1]).
+
+%%
+%% Example:
+%% Different functions to capitalize a string
+%%
+
+%% BAD: makes unnecessary use of manual recursion
+recurse(S) ->
+    lists:reverse(recurse(S, [])).
+
+recurse([], Acc) ->
+    Acc;
+recurse([H | T], Acc) ->
+    NewAcc = [string:to_upper(H) | Acc],
+    recurse(T, NewAcc).
+
+%% GOOD: uses a fold instead to achieve the same result,
+%% but this time more safely, and with fewer lines of code
+fold(S) ->
```

Comment:
Lists map `lists:map(fun string:to_upper/1, S)` would make the good example better (and almost the BEST ;-)